src/Microsoft.AspNetCore.Razor.Language/BoundAttributeDescriptorComparer.cs

```diff
@@ -76,15 +75,7 @@ public virtual int GetHashCode(BoundAttributeDescriptor descriptor)
 
             var hashCodeCombiner = HashCodeCombiner.Start();
             hashCodeCombiner.Add(descriptor.Kind);
-            hashCodeCombiner.Add(descriptor.IsIndexerStringProperty);
-            hashCodeCombiner.Add(descriptor.IsEnum);
-            hashCodeCombiner.Add(descriptor.HasIndexer);
             hashCodeCombiner.Add(descriptor.Name, _stringComparer);
-            hashCodeCombiner.Add(descriptor.IndexerNamePrefix, _stringComparer);
-            hashCodeCombiner.Add(descriptor.TypeName, StringComparer.Ordinal);
```

Comment:
Should we keep TypeName?

Reply:
Keep in mind any time the hash is used, the equality check is also likely to be used. Hashing helps us reject things that are non-matches efficiently.

Reply:
`TypeName` in this context doesn't have a lot of value because it's the parameters `TypeName` and not something that could globally represents the current bound attribute descriptor